upgrading.md

```diff
@@ -0,0 +1,202 @@
+
+1. Create a brand-new WordPress, withoud the database.
+2. Copy the old WordPress files from the "/wp-content/uploads/" content to the new one.
+3. Create a new database with the old database information. The best way is using "mysqldump".
+4. Configure the wp-config.php with all the new data.
+5. Access the "/wp-admin/" page, and follow the upgrading process.
+
+With this way, WordPress will be able to maintain and update the contents in the database and be able to work with these contents in an updated version of WordPress.
+
+A WordPress with the default theme, and all the contents should now be available.
+
+A common problem found in these database restorations is the character encoding type. It is very likely that its not in UTF-8 (maybe in an ISO or ASCII "deprecated" format), so make sure that the character encoding is updated correctly.
+
+## Upgrading from WordPress 3.7 - 4.0
+
+Goals
+- WordPress: upgrade to WordPress 4.1
+- PHP: upgrade to PHP 5.6.20+
+- SQL: upgrade to MySQL 5.6 / MariaDB 10.0
+
+Losses:
+- Content: none
+- Plugins: probably yes
+- Themes: probably yes
+
+WordPress Versions <= 4.0 are compatible with PHP versions that are hardly available today. They can range from PHP 5.2 (or even earlier) to PHP 5.5. That is why the main goal will be to go to a version that is still easy to get on many operating systems.
+
+The same will happen with the database. It is very likely that there is a MySQL 5.5 or earlier. Depending on whether want to continue with MySQL or move to MariaDB, choose which way to go and migrate the database to a MySQL 5.6 or MariaDB 10.0.
+
+Note that WP-CLI is not available for PHP versions lower than PHP 5.6.20, so this process still must be done somewhat manually.
+
+As with any upgrade, the first thing to do is to make a backup copy.
+
+Remove all themes that are not active, leaving only the main theme. If there is a child theme active, please, maintain the child and parent.
+
+Install and activate the [Twenty Ten](https://wordpress.org/themes/twentyten/) theme and activate it. This theme works in all sites since WordPress 3.7.
+
+In the same way, delete all deactivated plugins (and, therefore, not working).
+
+Deactivate all left active plugins.
+
+Now, WordPress will have:
+-	Core: any version (between WordPress 3.7 and WordPress 4.0)
+-	Themes: Twenty Ten is active, and the main theme is deactivated.
+-	Plugins: all plugins that should be active are deactivated.
+
+At this point, overwrite the WordPress Core with [WordPress 4.1](https://wordpress.org/wordpress-4.1.zip), available in the [release list](https://wordpress.org/download/releases/). Install WordPress 4.1 major version or, if available and recommended, the latest 4.1.x minor version.
+
+Upgrade the systems up to PHP 5.6.20+ and MySQL 5.6.x or MariaDB 10.0.x. Please, do not install the newest major version.
```

Comment:
is there ambiguity between 5.6.20+ and not later version? alternate: "5.6.X (for X >= 20)", "5.6 (recommend 5.6.20+)"

Reply:
> is there ambiguity between 5.6.20+ and a say a 7.0+ version? alternate: "5.6.X (for X >= 20)", "5.6 (recommend 5.6.20+)"

There are some incompatibilities between 5.6 and 7: https://www.php.net/manual/en/migration70.incompatible.php

Reply:
All the versions presented here are related to what was compatible with each WordPress version at the time. The idea is having the best version for each product.